### ISSUE TITLE
UCRT based w64devkit

### ISSUE THREAD

**Issue by 2cls0f** [skeeto/w64devkit#42]
winlibs.com has offered UCRT based MinGW for ages. It's very nice if w64devkit could offer UCRT based versions alongside traditional MSVCRT based versions. Thank you.

**Comment by skeeto** [comment]
Just like #44, I don't object to the idea as another variant, but since 
it's not something I expect I will ever personally need, I'm unlikely to 
build and maintain it. (Ideally the software I write will ultimately link 
with neither MSVCRT nor UCRT anyway.)

As far as I know, the only advantage for UCRT over MSVCRT is better MSVC 
compatibility, both API and ABI. Otherwise it's still just the same old 
crummy libc as MSVCRT: quirky behaviors, translating text streams, 
standard C functions are narrow API only, etc. The cost is reduced Windows 
compatibility and a more complex linkage situation (api sets, lots of 
DLLs).

**Comment by 2cls0f** [comment]
> Just like #44, I don't object to the idea as another variant, but since it's not something I expect I will ever personally need, I'm unlikely to build and maintain it. (Ideally the software I write will ultimately link with neither MSVCRT nor UCRT anyway.) As far as I know, the only advantage for UCRT over MSVCRT is better MSVC compatibility, both API and ABI. Otherwise it's still just the same old crummy libc as MSVCRT: quirky behaviors, translating text streams, standard C functions are narrow API only, etc. The cost is reduced Windows compatibility and a more complex linkage situation (api sets, lots of DLLs).

Modern Windows supports UTF-8 natively, UCRT is required, normal MSVCRT doesn't work.

https://stackoverflow.com/questions/3298569/difference-between-mbcs-and-utf-8-on-windows

**Comment by skeeto** [comment]
Before I wrote my last message I had searched for UCRT-unique features, 
including UTF-8 support, in case there was anything I missed. I found an R 
thread suggesting UCRT would make UTF-8 easier, but that's it. I could 
find no mention of UTF-8 in the official UCRT documentation.

So I just wrote a test using Visual Studio 2022, compiled like so:

    cl /utf-8 /MD test.c

The /utf-8 is because the source file is UTF-8, and /MD so I could verify 
it's linked against UCRT (yes). I made a file "π.txt", tried it with 
fopen(), tried printing the file name, and tried passing "π" as an 
argument. Everything behaved exactly as MSVCRT: not useful. If it's 
because I need to do something special in my program to access it, that's 
no better than MSVCRT where this is already the case. (Besides, that won't 
help with the already-destroyed argv.)

If I use w64devkit to compile, and link my "libwinsane.o" (embeds a UTF-8 
manifest, and some other startup configuration), this same program behaves 
just as it would on Linux, while linked against MSVCRT and without source 
changes. If UCRT worked like this, I'd be interested!

    cc libwinsane.o test.c

Details on libwinsane.o here:
https://nullprogram.com/blog/2021/12/30/

Your Stack Overflow link doesn't talk about UCRT, and indeed predates it 
by several years, so I don't understand the connection.

As far as I can tell, UTF-8 support is no better when using UCRT.
